### PR TITLE
Parametrize some TestAutogradFunctional tests to use logging_tensor

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73854
* #73853
* #73852

Remaining tests can be parametrized as well once this PR is finalized

Differential Revision: [D34703584](https://our.internmc.facebook.com/intern/diff/D34703584)

### CODE REVIEW

**Comment by soulitzer** [comment]
@albanD this might be ready for review. It's actually a bit tricky to convert to using `make_tensor` because there are some use cases that aren't covered: mainly `torch.ones`, `torch.zeros`, `torch.tensor`. If we simply parametrize over an argument of `make_tensor`, we're ignoring calls into `torch.ones` etc. (but maybe that is okay because most of the autograd stuff happens on the main input which is usually `torch.rand`). I'd like to avoid the hack of setting low/high to ones though. 

It seems like if I wanted to modify the behavior of calls into `torch.ones` AND use make_tensor, but I only have a single argument to parametrize over, I'd use the namespace approach that I use now. That seems to defeat the purpose of using `make_tensor` at all though since I think the main utility of `make_tensor` currently is setting different low/high ranges and for different dtypes.

`make_tensor` seems only useful to introduce if we eventually plan to parametrize over dtype (not too likely anyway since we mostly care about float32/float64 for autograd).

**Comment by soulitzer** [comment]
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34703584).

File: test/autograd/test_functional.py [hunk 1]
```diff
@@ -8,7 +9,45 @@
...
+
+# Utilities for parametrizing the tensor constructors used in autograd tests
+base_ctors_dict = {
+    "ones": torch.ones,
+    "zeros": torch.zeros,
+    "randn": torch.randn,
+    "rand": torch.rand,
+    "tensor": torch.tensor,
+    # TODO: maybe move somewhere so other tests can also use
+    # TODO: according to https://pytorch.org/cppdocs/notes/tensor_creation.html
```
Comment (albanD) [inline test/autograd/test_functional.py:23]: Do we need these other functions? Is that really a TODO?

File: test/autograd/test_functional.py [hunk 2]
```diff
@@ -8,7 +9,45 @@
 
 from torch.testing._internal.common_cuda import TEST_CUDA
 from torch.testing._internal.common_utils import (
-    TestCase, run_tests, gradcheck, gradgradcheck, parametrize, instantiate_parametrized_tests)
+    TestCase, run_tests, subtest, gradcheck, gradgradcheck, parametrize, instantiate_parametrized_tests)
+from torch.testing._internal.logging_tensor import LoggingTensor
+
+# Utilities for parametrizing the tensor constructors used in autograd tests
```
Comment (albanD) [inline test/autograd/test_functional.py:15]: I feel like these could go into a separate file.
But ok for now indeed.

**Review by albanD — APPROVED** [review]
Ok

File: test/autograd/test_functional.py (hunk 1 above)
Comment (albanD) [reply test/autograd/test_functional.py:23]: I'm fine with adding them as needed. And it is not really a TODO to be exhaustive here.

**Pull request closed without merging**